### PR TITLE
🚧Improve release workflow

### PR DESCRIPTION
**Pull request by kselena**
<h1 style="font-weight:bold" align="center">Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: feature/<issue-num>-<description>
      Example: feature/123-my-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_development_** branch with the name **_develop_**.

- [x] The **_base(target)_** branch for this pull request is a **_development_** branch with the name **_develop_**.
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

💡For more information on branching, refer to the project's [branching documentation](https://github.com/KinsonDigital/.github/blob/master/docs/BranchingFullGitFlow.md).

- [x] The associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] The pull request title matches the title of the associated issue.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] The assigned project matches the project of the associated issue.

- [x] The assigned milestone matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_develop_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: .github/workflows/add-new-issue-to-project.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+on:
+  issues:
+    types: opened
+
+
+jobs:
+  add_new_issue_to_project:
+    name: Add New Issue
+    uses: KinsonDigital/Infrastructure/.github/workflows/add-issue-to-project.yml@v8.2.0
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/add-new-issue-to-project.yml:12]: Updated to latest version

```suggestion
    uses: KinsonDigital/Infrastructure/.github/workflows/add-issue-to-project.yml@v9.0.0
```

File: .github/workflows/add-new-issue-to-project.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    with:
+      org-name: "${{ vars.ORGANIZATION_NAME }}"
+      org-project-name: "${{ vars.ORG_PROJECT_NAME }}"
+      project-name: "${{ vars.PROJECT_NAME }}"
+    secrets:
+      cicd-pat: ${{ secrets.CICD_REST_API }}
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/add-new-issue-to-project.yml:18]: Convert to different token:

```suggestion
      cicd-pat: ${{ secrets.CICD_TOKEN }}
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -7,98 +8,63 @@ defaults:
...
+    needs: determine_build_config
+    uses: KinsonDigital/Infrastructure/.github/workflows/dotnet-action-release.yml@v8.1.0
+    with:
+      project-name: "${{ vars.PROJECT_NAME }}"
+      release-type: "${{ inputs.release_type }}"
+      run-branch: "${{ github.ref_name }}"
+      net-sdk-version: "${{ vars.NET_SDK_VERSION }}"
+      relative-release-notes-dir-path: "${{ vars.SCRIPT_RELATIVE_DIR_PATH }}"
+      release-notes-file-name-prefix: "${{ vars.RELEASE_NOTES_FILE_NAME_PREFIX }}"
+      build-config: "${{ needs.determine_build_config.outputs.build-config }}"
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/release.yml:66]: Update workflow input value to:

```suggestion
      build-config: ${{ inputs.release-type == 'Production' && 'Release' || 'Debug' }}
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -7,98 +8,63 @@ defaults:
...
+          # If the branch that the workflow is running on is the required branch for the release, do a release build
+          if ( "${{ github.base_ref }}" -eq "main") {
+            "build-config=Release" >> $env:GITHUB_OUTPUT;
+            echo "Release build created.";
+          } else { # Any other branch than the release branch, do a debug build
+            "build-config=Debug" >> $env:GITHUB_OUTPUT;
+            echo "Debug build created.";
+          }
+
+
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/release.yml:54]: Remove job:

```suggestion
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -7,98 +8,63 @@ defaults:
...
+  perform_release:
+    name: Performing ${{ inputs.release_type }} Release of ${{ vars.PROJECT_NAME }} (${{ needs.determine_build_config.outputs.build-config }})
+    needs: determine_build_config
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/release.yml:57]: Remove needs from job:

```suggestion
```

File: .github/workflows/sync-bot.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            3. Event Type - set to issue event type
+            4. GitHub token
+          #>
+          deno run `
+            --allow-net `
+            "$scriptUrl" `
+            "${{ vars.PROJECT_NAME }}" `
+            "$issueNumber" `
+            "issue" `
+            "${{ secrets.CICD_REST_API }}";
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/sync-bot.yml:51]: Update token reference:

```suggestion
            "${{ secrets.CICD_TOKEN }}";
```

File: .github/workflows/sync-issue-to-pr.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+          deno run `
+            --allow-net `
+            "$scriptUrl" `
+            "${{ vars.ORGANIZATION_NAME }}" `
+            "${{ vars.PROJECT_NAME }}" `
+            "${{ github.event.sender.login }}" `
+            "$prNumber" `
+            "$command" `
+            "${{ secrets.CICD_REST_API }}";
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/sync-issue-to-pr.yml:51]: Update token reference:

```suggestion
            "${{ secrets.CICD_TOKEN }}";
```

File: .github/workflows/sync-status-check.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            3. Event Type - set to pull request event type
+            4. GitHub token
+          #>
+          deno run `
+            --allow-net `
+            "$scriptUrl" `
+            "${{ vars.PROJECT_NAME }}" `
+            "$prNumber" `
+            "pr" `
+            "${{ secrets.CICD_REST_API }}";
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/sync-status-check.yml:51]: Update token reference:

```suggestion
            "${{ secrets.CICD_TOKEN }}";
```

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]
